xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_epg_types.h

```diff
@@ -101,8 +101,8 @@ extern "C" {
     int           iParentalRating;     /*!< @brief (optional) parental rating */
     int           iStarRating;         /*!< @brief (optional) star rating */
     bool          bNotify;             /*!< @brief (optional) notify the user when this event starts */
-    int           iSeriesNumber;       /*!< @brief (optional) series number */
-    int           iEpisodeNumber;      /*!< @brief (optional) episode number */
+    int           iSeriesNumber;       /*!< @brief (optional) series number. Set to "0" for specials/pilot. For 'invalid' set to -1  */
+    int           iEpisodeNumber;      /*!< @brief (optional) episode number. For 'invalid' set to -1  */
     int           iEpisodePartNumber;  /*!< @brief (optional) episode part number */
```

Comment:
Can you also add for part number?

Reply:
Where is iEpisodePartNumber documented that 0 isn't invalid?   Also I don't think it is even displayed and the tests on it for series test are all > 0

I do have reservations that  iEpisodeNumber needs to -1 since this should be reserved for values where the guilib functions should display it and it doesn't with 0.

Reply:
It is not documented like that yet. This is what we are defining. In the other PR I have asked Ronie to update the series test also to match these values.

The idea with @ronie’s original PR is that it should be possible to set the values to any positive integer. Which makes sense. I have seen many pilots there are episode 0, e.g. S01E00.

Reply:
Ronie will make a change to the other PR so episode will be `>= 0`.

Reply:
Pilots as S>0E00 sounds like an afterthought on the TVDB and I have never seen it in the wild on TV.  In any case I was  ignoring that discussion   It is less clear why iEpisodePart number needs to be -1  I have often seen EPG with parts 1/3, 1/8 etec but  what is the use case for 0/5?  (the limit isn't part of Kodi spec but I put it in for this question.

Reply:
Episode part number is supposed to be used for when a single episode get's split for some reason. I.e. and news programme in between. Depending on your country they may be use part 1 and 2 or they may use part 0 and 1. 

We don't use this currently in Kodi but in the future if we did -1 for invalid would be correct.

Reply:
I have had been told not to commit for future undocumented possibilities and since your use case would likely require a code update https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/epg/EpgInfoTag.cpp#L646 I still am reluctant to include it as documentation

Reply:
The code exists already. If we are going to change every PVR addon to support the new invalid values I would rather do it once ;)

I have asked Ronie to update EpgInfoTag L646.

Reply:
Ok I will make the documentation chance but you have turned this from a cosmetic change to a breaking change since all the PVR addons  that haven't changed to -1 will tag all EPG items as IsSeries() true

Reply:
Yes, but I will update all the addons to set defaults as -1. Can merge this and Ronnie's PR once I have them ready.

Reply:
The issue isn't the default.  I looked at this when @ksooo said he didn't have time and if you don't know whether the backend supports E0  Using pvr.vuplus as an example https://github.com/kodi-pvr/pvr.vuplus/blob/Matrix/src/enigma2/data/RecordingEntry.cpp#L204

  left.iSeriesNumber = m_seasonNumber;
  left.iEpisodeNumber = m_episodeNumber;

you need to add

if   ( left.iSeriesNumber == 0 && left.iEpisodeNumber ==0 )
{
  left.iSeriesNumber = -1;
  left.iEpisodeNumber = -1;
}

That  still leaves open the use case that you suggested with the >= testing that the E0 could be valid now so S0 E0 could be a special.  Only the maintainer would know that.

Reply:
I don't follow. In this case you would just set the default in BaseEntry.h to -1 instead of 0.

Reply:
And what does that do to https://github.com/kodi-pvr/pvr.vuplus/blob/0a9d2c0efdd336e579dc988d171a20244b666632/src/enigma2/extract/ShowInfoExtractor.cpp#L63  and 77

That one will be easier then others too since we can see how they are formed.  If this is all done on the backend you don't

Reply:
If what you mean is that the solution is different in each addon, agreed.

Reply:
Here's the PR for vuplus: https://github.com/kodi-pvr/pvr.vuplus/pull/275

As each PR is created the maintainer can raise if a different change is needed. I agree there will be addons that require more work. But this needs to be done regardless.